pkgs/top-level/all-packages.nix

```diff
@@ -29380,9 +29380,13 @@ with pkgs;
     # Upstream thread: https://code.videolan.org/videolan/vlc/-/issues/26389
     libcaca = null;
   };
-
   vlc_qt5 = vlc;
 
+  vlc_wayland = vlc.override {
+    withQt5 = true;
+    waylandSupport = true;
+  };
+
```

Comment:
This shouldn't be done unless necessary. wayland support is not complete but it is better than nothing.